core/src/main/scala/pureconfig/ConfigCursor.scala

```diff
@@ -103,27 +102,65 @@ sealed trait ConfigCursor {
    */
   def asCollectionCursor: Either[ConfigReaderFailures, Either[ConfigListCursor, ConfigObjectCursor]] = {
     if (isUndefined) {
-      fail(KeyNotFound(path, location, Set()))
+      failed(KeyNotFound.forKeys(path, Set()))
     } else {
       val listAtLeft = asListCursor.right.map(Left.apply)
       lazy val mapAtRight = asObjectCursor.right.map(Right.apply)
       listAtLeft
         .left.flatMap(_ => mapAtRight)
-        .left.flatMap(_ => fail(WrongType(value.valueType, Set(ConfigValueType.LIST, ConfigValueType.OBJECT), location, path)))
+        .left.flatMap(_ => failed(WrongType(value.valueType, Set(ConfigValueType.LIST, ConfigValueType.OBJECT))))
     }
   }
 
+  /**
+   * Returns a failed `ConfigReader` result resulting from scoping a `FailureReason` into the context of this cursor.
+   *
+   * This operation is the easiest way to return a failure from a `ConfigReader`.
+   *
+   * @param reason the reason of the failure
+   * @tparam A the returning type of the `ConfigReader`
+   * @return a failed `ConfigReader` result built by scoping `reason` into the context of this cursor.
+   */
+  def failed[A](reason: FailureReason): Either[ConfigReaderFailures, A] =
+    Left(ConfigReaderFailures(failureFor(reason)))
+
+  /**
+   * Returns a `ConfigReaderFailure` resulting from scoping a `FailureReason` into the context of this cursor.
+   *
+   * This operation is useful for constructing `ConfigReaderFailures` when there are multiple `FailureReason`s.
```

Comment:
What's the reason? Because it is easier to scope multiple `FailureReason`s?

Reply:
Because you can easily do `listOfFailureReasons.map { fr => cur.failureFor(fr) }` and construct `ConfigReaderFailures` from a `List[ConfigReaderFailure]`. We could as well provide a `ConfigCursor#failed(FailureReason, FailureReason*)` variant, but I thought it would be more useful to provide the low-level way to "lift" a `FailureReason`. Do you think the API or this documentation should be improved?

Reply:
No I think it's clear, thank you.